sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -1025,29 +1021,70 @@ export class MessageReceiver extends LinkEntity {
         };
         return reject(translate(e));
       }, Constants.defaultOperationTimeoutInSeconds * 1000);
+
+      // This takes care of resolving the promise by waiting on the onSettled event
       this._deliveryDispositionMap.set(delivery.id, {
         resolve: resolve,
         reject: reject,
         timer: timer
       });
-      if (operation === DispositionType.complete) {
-        delivery.accept();
-      } else if (operation === DispositionType.abandon) {
-        const params: any = {
-          undeliverable_here: false
-        };
-        if (options.propertiesToModify) params.message_annotations = options.propertiesToModify;
-        delivery.modified(params);
-      } else if (operation === DispositionType.defer) {
-        const params: any = {
-          undeliverable_here: true
-        };
-        if (options.propertiesToModify) params.message_annotations = options.propertiesToModify;
-        delivery.modified(params);
-      } else if (operation === DispositionType.deadletter) {
-        delivery.reject(options.error || {});
+
+      try {
+        if (operation === DispositionType.complete) {
+          delivery.accept();
+        } else if (operation === DispositionType.abandon) {
+          const params: any = {
+            undeliverable_here: false
+          };
+          if (options.propertiesToModify) params.message_annotations = options.propertiesToModify;
+          delivery.modified(params);
+        } else if (operation === DispositionType.defer) {
+          const params: any = {
+            undeliverable_here: true
+          };
+          if (options.propertiesToModify) params.message_annotations = options.propertiesToModify;
+          delivery.modified(params);
+        } else if (operation === DispositionType.deadletter) {
+          delivery.reject(options.error || {});
+        }
+      } catch (err) {
```

Comment:
The try/catch being added here makes an assumption that the retryable errors that we want to catch are errors thrown by the `accept()`, `modified()` and `reject()` methods on the `delivery` object. This is not true.

The retryable errors that we want to look for are in the responses from the service. These responses come in the form of the `_onSettled` event. 

The other retryable error we want to catch is the timeout error which also will not be caught by this try/catch. This error is thrown a few lines above. Look for `Operation to settle the message has timed out.`

Reply:
It's possible that these operations could time out as well since they involve network calls, and those scenarios were being caught by the try catch. 
Since `onSettled` marks the end of overall operation, updated accordingly.

Reply:
These operations i.e. `accept()`, `modified()` and `reject()` methods on the `delivery` object are fire and forget methods. They are not async and do not have a timeout implemented internally. So, the try/catch wouldnt be catching the kind of errors we want to catch.